power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -308,87 +218,92 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         // set construction_complete for debug assertions
         construction_complete_ = true;
 #endif  // !NDEBUG
-        components_.set_construction_complete();
+        state_.components.set_construction_complete();
         // set component topo
         ComponentTopology comp_topo;
-        comp_topo.n_node = components_.template size<Node>();
+        comp_topo.n_node = state_.components.template size<Node>();
         // fill topology data
```

Comment:
i would like to move this to a separate file as well but i'd say that is out of scope of this PR cause it's already quite large